diff --git a/Makefile b/Makefile
index 303b2f7..9b3d683 100644
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ VSUBDIRS = hdl buildroot linux u-boot-xlnx
 
 VERSION=$(shell git describe --abbrev=4 --dirty --always --tags)
 LATEST_TAG=$(shell git describe --abbrev=0 --tags)
-UBOOT_VERSION=$(shell echo -n "PlutoSDR " && cd u-boot-xlnx && git describe --abbrev=0 --dirty --always --tags)
+UBOOT_VERSION=$(shell echo -n "ANTSDR " && cd u-boot-xlnx && git describe --abbrev=0 --dirty --always --tags)
 HAVE_VIVADO= $(shell bash -c "source $(VIVADO_SETTINGS) > /dev/null 2>&1 && vivado -version > /dev/null 2>&1 && echo 1 || echo 0")
 XSA_URL ?= http://github.com/analogdevicesinc/plutosdr-fw/releases/download/${LATEST_TAG}/system_top.xsa
 
@@ -38,8 +38,8 @@ $(error "      3] export VIVADO_VERSION=v20xx.x")
 	endif
 endif
 
-TARGET ?= pluto
-SUPPORTED_TARGETS:=pluto sidekiqz2
+TARGET ?= ant
+SUPPORTED_TARGETS:=pluto sidekiqz2 ant
 
 # Include target specific constants
 include scripts/$(TARGET).mk
@@ -59,7 +59,7 @@ endif
 
 ifeq ($(findstring $(TARGET),$(SUPPORTED_TARGETS)),)
 all:
-	@echo "Invalid `TARGET variable ; valid values are: pluto, sidekiqz2" &&
+	@echo "Invalid `TARGET variable ; valid values are: pluto, sidekiqz2,ant" &&
 	exit 1
 else
 all: clean-build $(TARGETS) zip-all legal-info
diff --git a/linux b/linux
--- a/linux
+++ b/linux
@@ -1 +1 @@
-Subproject commit 9dfba10b795d0004ae90f2ab29dac0197c8a3b3e
+Subproject commit 9dfba10b795d0004ae90f2ab29dac0197c8a3b3e-dirty
diff --git a/u-boot-xlnx b/u-boot-xlnx
--- a/u-boot-xlnx
+++ b/u-boot-xlnx
@@ -1 +1 @@
-Subproject commit 90401ce9ce029e5563f4dface63914d42badf5bc
+Subproject commit 90401ce9ce029e5563f4dface63914d42badf5bc-dirty
